tests/staking_info_test.go

```diff
@@ -0,0 +1,97 @@
+package tests
+
+import (
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	rewardcontract "github.com/klaytn/klaytn/contracts/reward/contract"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/klaytn/klaytn/governance"
+	"github.com/klaytn/klaytn/log"
+	"github.com/klaytn/klaytn/params"
+	"github.com/klaytn/klaytn/reward"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddressBookConnector(t *testing.T) {
+	log.EnableLogForTest(log.LvlCrit, log.LvlWarn)
+
+	fullNode, node, validator, chainId, workspace := newBlockchain(t)
+	defer os.RemoveAll(workspace)
+	defer fullNode.Stop()
+
+	var (
+		chain  = node.BlockChain().(*blockchain.BlockChain)
+		config = chain.Config()
+		txpool = node.TxPool()
+		db     = node.ChainDB()
+		gov    = governance.NewMixedEngine(config, db)
+
+		deployAddr  common.Address
+		deployBlock uint64
+	)
+
+	// Deploy AddressBook
+	{
+		sender := validator
+		signer := types.LatestSignerForChainID(chainId)
+
+		values := map[types.TxValueKeyType]interface{}{
+			types.TxValueKeyNonce:         sender.GetNonce(),
+			types.TxValueKeyAmount:        new(big.Int).SetUint64(0),
+			types.TxValueKeyGasLimit:      uint64(1e9),
+			types.TxValueKeyGasPrice:      big.NewInt(25 * params.Ston),
+			types.TxValueKeyHumanReadable: false,
+			types.TxValueKeyFrom:          sender.GetAddr(),
+			types.TxValueKeyData:          common.FromHex(rewardcontract.AddressBookBin),
+			types.TxValueKeyCodeFormat:    params.CodeFormatEVM,
+			types.TxValueKeyTo:            (*common.Address)(nil),
+		}
+
+		tx, err := types.NewTransactionWithMap(types.TxTypeSmartContractDeploy, values)
+		require.Nil(t, err)
+
+		err = tx.SignWithKeys(signer, sender.GetTxKeys())
+		require.Nil(t, err)
+
+		err = txpool.AddLocal(tx)
+		require.True(t, err == nil || err == blockchain.ErrAlreadyNonceExistInPool, "err=%v", err)
+
+		deployAddr = crypto.CreateAddress(sender.Addr, sender.Nonce)
+		sender.AddNonce()
+
+		receipt := waitReceipt(chain, tx.Hash())
+		require.NotNil(t, receipt)
+		require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status)
+
+		_, _, deployBlock, _ = chain.GetTxAndLookupInfo(tx.Hash())
+		t.Logf("AddressBook deployed at block=%2d", deployBlock)
+	}
+
+	// Temporarily use the newly deployed address in addressBookConnector
+	oldAddr := common.HexToAddress(rewardcontract.AddressBookContractAddress)
+	reward.SetTestAddressBookAddress(deployAddr)
```

Comment:
Is this the newly deployed address, not validator?

Reply:
I think it's a validator address from newBlockchain(t). We cannot use `rewardcontract.AddressBookContractAddress` for it?

Reply:
1. It's not validator address. It's the deployed contract address which is calculated from sender address and sender nonce (https://ethereum.org/en/developers/docs/accounts/#contract-accounts).
2. We cannot use `rewardcontract.AddressBookContractAddress = 0x000..400` because contract is not there. Cypress and Baobab has allocated AddressBook contract at `0x400` address in their genesis. But the test blockchain created by `newBlockchain()` don't. Therefore I've deployed a new AddressBook and overriden addressBookConnector.

Reply:
I got it. Thanks for the explanation.